cocos2d/core/components/CCLayout.js

```diff
@@ -357,7 +357,10 @@ var Layout = cc.Class({
             this.node.setContentSize(this._layoutSize);
         }
 
-        this.node.on('size-changed', this._resized, this);
+        if(this.resizeMode !== ResizeMode.CONTAINER) {
```

Comment:
because in ResizeMode.Container, the layout will change node's size, so there is no need to cause layout to dirty...